[CT-3374] [Bug] test_utils.TestCurrentTimestamp.test_current_timestamp_matches_utc is failing for the MacOS runner

### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

The MacOS runner system clock appears to be a few minutes behind. When running `test_utils.TestCurrentTimestamp.test_current_timestamp_matches_utc`, the database time winds up being more than two minutes ahead of the system clock, despite being sampled first. This appears to only affect the MacOS runner.

### Expected Behavior

The check should work across all operating systems, or should at least account for this anomaly.

### Steps To Reproduce

Observe the test results in GHA.

### Relevant log output

```shell
___________ TestCurrentTimestamp.test_current_timestamp_matches_utc ____________
 See https://docs.pytest.org/en/latest/deprecations.html#configuring-hook-specs-impls-using-markers
[gw2] darwin -- Python 3.8.18 /Users/runner/work/dbt-snowflake/dbt-snowflake/.tox/integration-snowflake/bin/python

self = <test_utils.TestCurrentTimestamp object at 0x10a0f9a90>
current_timestamp = datetime.datetime(2023, 10, 31, 14, 32, 28, 79000, tzinfo=<UTC>)

    def test_current_timestamp_matches_utc(self, current_timestamp):
        sql_timestamp = current_timestamp
        now_utc = self.utcnow_matching_type(sql_timestamp)
        # Plenty of wiggle room if clocks aren't perfectly sync'd, etc
        tolerance = timedelta(minutes=1)
>       assert (sql_timestamp > (now_utc - tolerance)) and (
            sql_timestamp < (now_utc + tolerance)
        ), f"SQL timestamp {sql_timestamp.isoformat()} is not close enough to Python UTC {now_utc.isoformat()}"
E       AssertionError: SQL timestamp 2023-10-31T14:32:28.079000+00:00 is not close enough to Python UTC 2023-10-31T14:29:09.986959+00:00
E       assert (datetime.datetime(2023, 10, 31, 14, 32, 28, 79000, tzinfo=<UTC>) > (datetime.datetime(2023, 10, 31, 14, 29, 9, 986959, tzinfo=datetime.timezone.utc) - datetime.timedelta(seconds=60)) and datetime.datetime(2023, 10, 31, 14, 32, 28, 79000, tzinfo=<UTC>) < (datetime.datetime(2023, 10, 31, 14, 29, 9, 986959, tzinfo=datetime.timezone.utc) + datetime.timedelta(seconds=60)))

/Users/runner/work/dbt-snowflake/dbt-snowflake/.tox/integration-snowflake/lib/python3.8/site-packages/dbt/tests/adapter/utils/test_current_timestamp.py:44: AssertionError
```


### Environment

```markdown
- OS: MacOS
- Python: py38
- dbt: 1.5 and later
```


### Which database adapter are you using with dbt?

other (mention it in "Additional Context")

### Additional Context

Affects `dbt-snowflake`, `dbt-bigquery`, and `dbt-redshift`

--- a/.changes/unreleased/Under the Hood-20231111-175350.yaml
+++ b/.changes/unreleased/Under the Hood-20231111-175350.yaml
@@ -1,0 +1,7 @@
+kind: Under the Hood
+body: Fix test_current_timestamp_matches_utc test; allow for MacOS runner system clock
+  variance
+time: 2023-11-11T17:53:50.098843-05:00
+custom:
+  Author: mikealfare
+  Issue: "9057"

--- a/tests/adapter/dbt/tests/adapter/utils/test_current_timestamp.py
+++ b/tests/adapter/dbt/tests/adapter/utils/test_current_timestamp.py
@@ -40,7 +40,8 @@ class BaseCurrentTimestamp:
         sql_timestamp = current_timestamp
         now_utc = self.utcnow_matching_type(sql_timestamp)
         # Plenty of wiggle room if clocks aren't perfectly sync'd, etc
-        tolerance = timedelta(minutes=1)
+        # The clock on the macos image appears to be a few minutes slow in GHA, causing false negatives
+        tolerance = timedelta(minutes=5)
         assert (sql_timestamp > (now_utc - tolerance)) and (
             sql_timestamp < (now_utc + tolerance)
         ), f"SQL timestamp {sql_timestamp.isoformat()} is not close enough to Python UTC {now_utc.isoformat()}"

